supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/GenomicsFactory.scala

```diff
@@ -47,7 +58,75 @@ case class GenomicsFactory(applicationName: String, authMode: GoogleAuthMode, en
       genomics.projects().operations().get(job.jobId).buildHttpRequest()
     }
 
-    override def runRequest(createPipelineParameters: CreatePipelineParameters, jobLogger: JobLogger) = {
+    override def runRequest(createPipelineParameters: CreatePipelineParameters, jobLogger: JobLogger): HttpRequest = {
+
+      def projectMetadataRequest(vpcConfig: VirtualPrivateCloudConfiguration): IO[HttpRequest] = {
+        val workflowOptions = createPipelineParameters.jobDescriptor.workflowDescriptor.workflowOptions
+
+        val googleCredentialOption = vpcConfig.auth.apiClientGoogleCredential((key: String) => workflowOptions.get(key).get)
+
+        googleCredentialOption match {
+          case None => IO.raiseError(new RuntimeException(s"Unable to find Google Credential for auth `${vpcConfig.auth.name}`."))
+          case Some(googleCredential) => IO {
+            val auth = googleCredential.createScoped(ResourceManagerAuthScopes)
+
+            val cloudResourceManagerBuilder = new CloudResourceManager
+              .Builder(GoogleAuthMode.httpTransport, GoogleAuthMode.jsonFactory, auth)
+              .build()
+
+            val project = cloudResourceManagerBuilder.projects().get(createPipelineParameters.projectId)
+
+            project.buildHttpRequest()
+          }
+        }
+      }
+
+
+      def projectMetadataResponseToLabels(httpResponse: HttpResponse): IO[ProjectLabels] = {
+        IO.fromEither(decode[ProjectLabels](httpResponse.parseAsString())).handleErrorWith {
+          e => IO.raiseError(new RuntimeException(s"Failed to parse labels from project metadata response from Google Cloud Resource Manager API. " +
```

Comment:
Since you're doing a `fromEither` already, could you add this context with a `leftMap` on the `Either` result, rather than having to create a whole new Exception instance?

Also (and this could be moot if you're able to do the above suggestion), I wouldn't make *this* a `RuntimeException` either.